src/app/modules/node/reducers.ts

```diff
@@ -16,6 +16,12 @@ export interface NodeState {
   checkAuthError: null | Error;
   isFetchingNodeInfo: boolean;
   fetchNodeInfoError: null | Error;
+  
+  editingNodeField: EditingNodeField;
```

Comment:
This field also has the same problems as `isChangingPassword` but it should be easier to move to component state because it displays the drawer first and doesn't `requirePassword` until the Continue button is clicked. I'd still have the same problem with automatically closing the drawer, but I can get around this using the condition in `componentWillUpdate`.

Reply:
I removed this field and put it into state. You were right about this change. The code is easier to follow.